### ISSUE TITLE
alternatives to dhtcrawler2

### ISSUE THREAD

**Issue by milahu** [btdig/dhtcrawler2#5]
## magnetico

- https://github.com/boramalper/magnetico
- archived
- 3K github stars
- last commit 2022
- written in: Go

<blockquote>

magnetico: Autonomous (self-hosted) BitTorrent DHT search engine suite.

magnetico is the first autonomous (self-hosted) BitTorrent DHT search engine suite that is designed for end-users. The suite consists of two packages:

- magneticod: Autonomous BitTorrent DHT crawler and metadata fetcher.
- magneticow: Lightweight web interface for magnetico.

Both programs, combined together, allows anyone with a decent Internet connection to access the vast amount of torrents waiting to be discovered within the BitTorrent DHT space, without relying on any central entity.

magnetico liberates BitTorrent from the yoke of centralised trackers & web-sites and makes it truly decentralised. Finally!

</blockquote>

### magnetico forks

- https://github.com/tgragnato/magnetico - 30 github stars, last commit 2024
- https://github.com/ireun/magnetico - 15 github stars, last commit 2022
- https://github.com/ez-me/magnetico - 10 github stars, last commit 2023

### magnetico issues

https://github.com/boramalper/magnetico/issues

#### use postgreSQL

- https://github.com/boramalper/magnetico/issues/264
- https://github.com/boramalper/magnetico/issues/280
   - switch to postgres database increases performance by about 4 times on my server

#### DHT database dumps <a href="#dht-database-dumps">#</a>

<a name="dht-database-dumps"></a>

todo: these should be sqlite files, so they can be used directly, without unpacking

- https://github.com/boramalper/magnetico/issues/218
   - https://tnt.maiti.info/dhtd/ - last release: 2025-12-30
   - https://antipoul.fr/dhtd/ - last release: 2025-04-17
   - https://www.kescher.at/magneticod-db - last release: 2021-05-19
   - https://framagit.org/Glandos/magnetico_merge - merge sqlite databases

#### DHT database sharing

- https://github.com/boramalper/magnetico/issues/187

#### distributed crawling

- https://github.com/boramalper/magnetico/issues/155
- https://github.com/boramalper/magnetico/issues/106
- https://github.com/boramalper/magnetico/issues/23

## shiyanhui/dht

- https://github.com/shiyanhui/dht
- written in: Go
- 3K github stars
- last commit 2020

## bitmagnet

- https://github.com/bitmagnet-io/bitmagnet
- written in: Go
- 2K github stars
- last commit 2024

## dhtbay

- https://github.com/FlyersWeb/dhtbay
- written in: JavaScript
- 100 github stars
- last commit 2018

## dhtsearch

- https://github.com/felix/dhtsearch
- written in: Go
- 50 github stars
- last commit 2020

## hamster

- https://github.com/picotorrent/hamster
- written in: Go
- 5 github stars
- last commit 2022

**Comment by tuxayo** [comment]
Is there any public instance of any of these? (those who have a web interface at least)

**Comment by ralyodio** [comment]
https://www.npmjs.com/package/torrent-search-api-for-barbaroussa

this is the only search module I found that works and it only works on 1337x

**Comment by milahu** [comment]
> Is there any public instance of any of these?

not really. only the usual torrent search engines...
[btdig.com](https://btdig.com/) [torrentz2.nz](https://torrentz2.nz/) [bitsearch.to](https://bitsearch.to/) [snowfl.com](https://snowfl.com/) [damag.net](https://damag.net/) [1337x.to](https://1337x.to/)

maybe [magnetico](https://github.com/tgragnato/magnetico) and [bitmagnet](https://github.com/bitmagnet-io/bitmagnet) are used as scrapers in the backend

**Comment by milahu** [comment]
## warning: illegal torrents

> DHT database dumps

warning: these torrent indexes are uncensored
so they can contain illegal content like childporn (CSAM)

[police are monitoring illegal torrents, so they can break into your home and steal your hardware](https://github.com/milahu/alchi/blob/master/deutsch/feedback/knast-2024/2025-02-01.kinderpornos-einlassung.md)
at least here in germany, maybe the situation is better in other jurisdictions

even if some years later, a judge decides
that your download was not intentional (negligent) and thereby legal
you still lose your hardware for some years
and of course you get no compensation
because you cannot prove your damage...

sharing metadata of illegal content is legal
but downloading (and especially uploading) illegal content is risky

if police would be playing a fair game
then they would publish blocklists of hashed infohashes of illegal content
so we can avoid negligent downloading of illegal content
but since when are police playing a fair game...?
police use CSAM as a pretext to destroy innocent people
because "we have to save the children at all cost"
(just another case of "the cure is worse than the disease")

**Comment by APT37** [comment]
What exactly is this about? It's a well-known fact that the DHT in unfiltered unless someone specifically implements a blacklist system on their node. The way you phrase your post makes it sound like you got caught (accidentally? :suspect:) downloading CSAM, not realizing that anyone in the swarm can potentially log your activity. 👏

**Comment by milahu** [comment]
> it sound like you got caught (accidentally?) downloading CSAM

yes, accidentally, just read my [statement to the court](https://github.com/milahu/alchi/blob/master/deutsch/feedback/knast-2024/2025-02-01.kinderpornos-einlassung.md)

under a rule of law, there would be the principle of [presumption of innocence](https://en.wikipedia.org/wiki/Presumption_of_innocence)
until my enemies actually prove that i am guilty of **intentional** sharing of illegal content
but when it comes to children, [most peoples brains shut down](https://en.wikipedia.org/wiki/Neuroticism)
and they switch into the "we have to destroy the childfucker" mode
(what a useful tool to destroy political enemies...)

im just saying that it may have serious consequences to download uncensored torrents
because even if the torrent has legal filenames like `Just.Some.Movie.mp4`
its content can be illegal and get you in trouble
(except you are rich, then you can buy a lawyer...)

**Comment by Glandos** [comment]
> sharing metadata of illegal content is legal

OK, then it's all what we got here: metadata. It's true that CSAM is really a hot topic, politically, but a simple warning message should be enough: "Do not attempt to blindly download torrent in this index, it's unfiltered, and potentially illegal"
